galasa-ui/src/components/users/UsersTable.tsx

```diff
@@ -6,19 +6,24 @@
 "use client";
 import { UserData } from '@/generated/galasaapi';
 import React, { useEffect, useState } from 'react';
-import { Loading } from "@carbon/react";
+import { Loading, Button, DataTable } from "@carbon/react";
 import { Table, TableHead, TableRow, TableBody, TableCell, TableHeader, TableContainer, TableToolbarSearch, TableToolbarContent } from '@carbon/react';
-import styles from "@/styles/UsersList.module.css";
-import { DataTable } from '@carbon/react';
 import ErrorPage from '@/app/error/page';
 import { TableRowProps } from '@carbon/react/lib/components/DataTable/TableRow';
 import { TableHeadProps } from '@carbon/react/lib/components/DataTable/TableHead';
 import { TableBodyProps } from '@carbon/react/lib/components/DataTable/TableBody';
+import { Edit } from '@carbon/icons-react';
+import styles from "@/styles/UsersList.module.css";
+import Link from 'next/link';
 
 export const dynamic = "force-dynamic";
 
-interface UsersListSectionProps {
-  usersListPromise: Promise<UserData[]>;
+type UsersListPromiseType = Promise<UserData[]>;
+type CurrentUserPromiseType = Promise<UserData>;
```

Comment:
I don't think we need these type aliases - we could just use actual types in the interface